agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -250,18 +250,25 @@ func (state *DockerTaskEngineState) AddContainer(container *api.DockerContainer,
 	}
 
 	if container.DockerID != "" {
+		// Update the container id to the state
```

Comment:
Is there a unit test covering the `if` and `else if` code paths?

Reply:
added.